axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineServiceImpl.java

```diff
@@ -130,12 +133,59 @@ public void computeProductInformation(SaleOrderLine saleOrderLine, SaleOrder sal
     fillComplementaryProductList(saleOrderLine);
   }
 
+  @Override
+  public void computePricingScale(SaleOrderLine saleOrderLine, SaleOrder saleOrder)
+      throws AxelorException {
+
+    // It is supposed that only one pricing match those criteria (because of the configuration)
+    // Having more than one pricing matched may result on a unexpected result
+    Pricing pricing =
+        pricingService
+            .getRandomPricing(
+                saleOrder.getCompany(),
+                saleOrderLine.getProduct(),
+                saleOrderLine.getProduct().getProductCategory(),
+                SaleOrderLine.class.getSimpleName(),
+                null)
+            .orElse(null);
+    if (pricing != null) {
+      PricingComputer pricingComputer =
+          PricingComputer.of(
+                  pricing, saleOrderLine, saleOrderLine.getProduct(), SaleOrderLine.class)
+              .putInContext("saleOrder", EntityHelper.getEntity(saleOrder));
+      pricingComputer.apply();
+    }
+  }
+
+  @Override
+  public boolean hasPricingLine(SaleOrderLine saleOrderLine, SaleOrder saleOrder)
+      throws AxelorException {
+    Pricing pricing =
+        pricingService
+            .getRandomPricing(
+                saleOrder.getCompany(),
+                saleOrderLine.getProduct(),
+                saleOrderLine.getProduct().getProductCategory(),
+                SaleOrderLine.class.getSimpleName(),
+                null)
+            .orElse(null);
+    if (pricing != null) {
```

Comment:
Is this possible to factorize some of the code here ? And maybe use Optional.ifPresent() ?